### PR TITLE
Project1- Using gspread library to populate a google sheet

### PR DESCRIPTION
**Pull request by LogicAL007**

### CODE REVIEW

File: my_gspread.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+def populate_worksheet(Ecom):
+    worksheet = get_or_create_worksheet(spreadsheet, Ecom, rows=df.shape[0]+1, cols=df.shape[1])
+    worksheet.update([df.columns.values.tolist()] + df.values.tolist())
+
+
+populate_worksheet("Ecommerce store")
+worksheet_list = spreadsheet.worksheets()
+print(worksheet_list)
+# worksheet= spreadsheet.worksheet("Ecommerce store")
```
Comment (elseagle) [inline my_gspread.py]: remove redundant comments

File: omotosho-project-c92e4f4bc071.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (elseagle) [inline omotosho-project-c92e4f4bc071.json]: don't commit a credential file. Use .env with `decouple` or `python-dotenv` library. Ensure you add your .env to .gitignore

File: my_gspread.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import gspread
+import pandas as pd
+
+# create a connection with the service account
+gc = gspread.service_account("omotosho-project-c92e4f4bc071.json")
+df = pd.read_csv("fileasa.csv")
+spreadsheet = gc.open("Gspread practice")
+# sh = gc.create('A new spreadsheet')
+
+def get_or_create_worksheet(spreadsheet, Ecom , rows=1, cols=1):
```
Comment (elseagle) [inline my_gspread.py]: add docstring to explain what your code does

File: my_gspread.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    try:
+        worksheet = spreadsheet.worksheet(Ecom)
+    except:
+        worksheet = spreadsheet.add_worksheet(Ecom, rows, cols)
+    return worksheet
+
+def populate_worksheet(Ecom):
```
Comment (elseagle) [inline my_gspread.py]: some docstring or comment will help here

File: my_gspread.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    worksheet = get_or_create_worksheet(spreadsheet, Ecom, rows=df.shape[0]+1, cols=df.shape[1])
+    worksheet.update([df.columns.values.tolist()] + df.values.tolist())
+
+
+populate_worksheet("Ecommerce store")
+worksheet_list = spreadsheet.worksheets()
+print(worksheet_list)
```
Comment (elseagle) [inline my_gspread.py]: don't use print, checkout the `logging` library and replace all print statements with it

**Review by elseagle — CHANGES_REQUESTED** [review]
Good start @LogicAL007 - added some comments.

Kindly use descriptive titles for your PRs and commits. Also, could you add description so we can have a summary of what you've built? 

A readme markdown file to give detail insight to your project

File: my_gspread.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+#read the data to be loaded
+df = pd.read_csv("fileasa.csv")
+
+#open the spreadsheet
+spreadsheet = gc.open("Gspread practice")
+
+logging.basicConfig(level=logging.INFO)
+
+def get_or_create_worksheet(spreadsheet, Ecom , rows=1, cols=1):
```
Comment (elseagle) [inline my_gspread.py]: don't use title format for parameters "Ecom" should be "ecom"

File: my_gspread.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        logging.info("The worksheet was not found, a new one will be created")
+    return worksheet
+
+def populate_worksheet(Ecom):
+    '''
+    This function puts the header of a dataframe into the first row of a sheet followed by the values of a dataframe to the worsksheet that was created or already existing.
+    
+    :param Ecom: The title of the worksheet to populate.
+    '''
+    worksheet = get_or_create_worksheet(spreadsheet, Ecom, rows=df.shape[0]+1, cols=df.shape[1])
```
Comment (elseagle) [inline my_gspread.py]: can you comment on this line to help us know why you are doing `df.shape[0]+1` ? It is unclear to everyone else except , help us make it clear to anyone that picks your code

File: README.md [hunk 8]
```diff
@@ -1,6 +0,0 @@
-# Populating an excel sheet with Gspread library
```
Comment (tobaadesugba) [inline README.md:1]: Your readme is a little incomplete. Include details like the author of the project, the tools you used, the source of the data used, and other details you feel are important.

**Review by tobaadesugba — CHANGES_REQUESTED** [review]

**Review by elseagle — APPROVED** [review]

**Pull request merged**